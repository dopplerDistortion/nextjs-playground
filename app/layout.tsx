import '@/styles/global.css';
import {ThemeProvider} from '@/components/providers/theme-provider';
import {cn} from '@/lib/utils';
import {Nunito_Sans as FontSans} from 'next/font/google';
import {ThemeWrapper} from '@/components/providers/theme-wrapper';

export const fontSans = FontSans({
  subsets: ['latin'],
  variable: '--font-sans',
});

export default function RootLayout({children}: {children: React.ReactNode}) {
  return (
    <html lang="en" suppressHydrationWarning>
      <body
        className={cn(
          'min-h-screen bg-background font-sans antialiased',
          fontSans.variable,
        )}>
        <ThemeProvider
          attribute="class"
          defaultTheme="system"
          enableSystem
          disableTransitionOnChange>
          <ThemeWrapper>{children}</ThemeWrapper>
        </ThemeProvider>
      </body>
    </html>
  );
}
