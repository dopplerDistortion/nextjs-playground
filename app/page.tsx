'use client';
import {
  Card,
  CardContent,
  CardDescription,
  CardFooter,
  CardHeader,
  CardTitle,
} from '@/components/ui/card';
import {Input} from '@/components/ui/input';
import {Label} from '@/components/ui/label';
import {useState} from 'react';

export default function Home() {
  const [value, setValue] = useState('');

  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
    const keyValue = event.key;
    // Allow Backspace key
    if (keyValue === 'Backspace') {
      return;
    }
    // Only allow numbers
    if (!/^\d+$/.test(keyValue)) {
      event.preventDefault();
      return;
    }
    const newValue = event.currentTarget.value + keyValue;
    // Only allow numbers between 1 and 1000
    if (parseInt(newValue, 10) < 1 || parseInt(newValue, 10) > 1000) {
      event.preventDefault();
      return;
    }
    setValue(newValue);
  };

  const handleKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
    const keyValue = event.key;
    // Update value state if Backspace key is released
    if (keyValue === 'Backspace') {
      setValue(event.currentTarget.value);
    }
  };

  const convertToRoman = (num: number) => {
    const units = {
      I: 1,
      V: 5,
    };
    const tens = {
      X: 10,
      L: 50,
    };
    const hundreds = {
      C: 100,
      D: 500,
    };
    const thousands = {
      M: 1000,
    };
  };

  return (
    <div className="flex flex-col justify-center items-center h-screen p-8 sm:p-0">
      <Card>
        <CardHeader>
          <CardTitle>Roman Numeral Converter</CardTitle>
          <CardDescription>
            Type number between 1 to 1000 inclusive
          </CardDescription>
        </CardHeader>
        <CardContent>
          <Input
            type="number"
            placeholder="Enter a number between 1 and 1000"
            onKeyDown={handleKeyDown}
            onKeyUp={handleKeyUp}
            max="1000"
          />
        </CardContent>
        <CardContent>
          <CardTitle className="text-sm">Converted Value</CardTitle>
          <CardDescription>
            {value === '' ? 'Please type a value' : value}
          </CardDescription>
        </CardContent>
      </Card>
    </div>
  );
}
