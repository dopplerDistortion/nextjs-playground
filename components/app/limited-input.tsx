import {useState} from 'react';
import {Input} from '../ui/input';

type LimitedInputProps = {
  setValue: (value: string) => void;
};

export function LimitedInput({setValue}: LimitedInputProps) {
  const [value, setLocalValue] = useState('');

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const newValue = event.currentTarget.value;
    if (newValue === '') {
      setLocalValue(newValue);
      setValue(newValue);
      return;
    }
    if (!/^\d+$/.test(newValue)) {
      event.preventDefault();
      return;
    }
    if (parseInt(newValue, 10) >= 1 && parseInt(newValue, 10) <= 1000) {
      setLocalValue(newValue);
      setValue(newValue);
    } else {
      event.preventDefault();
      return;
    }
  };

  return (
    <Input
      data-testid="limited-input"
      value={value}
      onChange={handleChange}
      type="number"
      className="mt-2 placeholder:opacity-50"
      placeholder="Number between 1 and 1000"
      max={'1000'}
      min={'1'}
    />
  );
}
