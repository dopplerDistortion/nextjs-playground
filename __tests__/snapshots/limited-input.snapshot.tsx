import {LimitedInput} from '@/components/app/limited-input';
import {render} from '@testing-library/react';
import {useState} from 'react';

const DummyLimitedInput = () => {
  const [value, setValue] = useState('123');
  return <LimitedInput setValue={setValue} />;
};
it('renders separator unchanged', () => {
  const {container} = render(<DummyLimitedInput />);
  expect(container).toMatchSnapshot();
});
