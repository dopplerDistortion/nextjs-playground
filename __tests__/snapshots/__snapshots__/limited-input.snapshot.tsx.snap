// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders separator unchanged 1`] = `
<div>
  <input
    class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 mt-2 placeholder:opacity-50"
    data-testid="limited-input"
    max="1000"
    min="1"
    placeholder="Number between 1 and 1000"
    type="number"
    value=""
  />
</div>
`;
